File: client/src/dataUtils.js
// // file is deprecated, not in use
// const fetchLocationsData = async () => {
//     try {
//         const response = fetch('/static/locations.json')
//         if (!response.ok){
//             throw new Error('Failed to fetch locations data')
//         }
//         const data = await response.json()
//         return data;
//     } catch ( error ){
//         console.error(error);
//         return []
//     }
// }

// const getCityToAirportMap = () => { // returns object map of cities to airports
//     const cityToAirportMap = {};
//     for (const location of locationsData) {
//         cityToAirportMap[location.city] = location.airport;
//     }
//     return cityToAirportMap;
// };

// const getAirportToCityMap = () => { // returns object map of airports to cities
//     const airportToCityMap = {};
//     for (const location of locationsData) {
//         airportToCityMap[location.airport] = location.city;
//     }
//     return airportToCityMap;
// };

// const getLocations = () => { // returns array of locations (cities and airports)
//     return locationsData
// }


// fetchLocationsData().then(locationsData => {
//     const cityToAirportMap = getAirportToCityMap(locationsData)
//     const airportToCityMap = getCityToAirportMap(locationsData)
//     const locations = getLocations(locationsData)
// })

// export { getAirportToCityMap, getCityToAirportMap, getLocations }

File: server/data_utils.py
import json

def get_city_to_airport_map(): # <-- dictionary of cities to airports (key:value)
    with open('./static/locations.json', 'r') as json_file:
        locations = json.load(json_file)
    city_to_airport_map = {l['city']: l['airport'] for l in locations}
    return city_to_airport_map

def get_airport_to_city_map(): # <-- dictionary of airports to cities (key:value)
    with open('./static/locations.json', 'r') as json_file:
        locations = json.load(json_file)
    airport_to_city_map = {l['airport']: l['city'] for l in locations}
    return airport_to_city_map

def get_airport_list(): # <-- returns list of airports
    with open('./static/locations.json', 'r') as json_file:
        data = json.load(json_file)
        return [l['airport'] for l in data['locations']]